pkg/producer/kafka_test.go

```diff
@@ -31,14 +31,14 @@ import (
 
 var (
 	// Hard coding the kafka msg in bytes. Need to figure out a way to generate this.
-	msg1InBytes = []byte{0x56, 0x10, 0x1, 0x18, 0xd2, 0x9, 0x32, 0x8, 0x31, 0x30,
+	msg1InBytes = []byte{0x10, 0x1, 0x18, 0xd2, 0x9, 0x32, 0x8, 0x31, 0x30,
 		0x2e, 0x30, 0x2e, 0x30, 0x2e, 0x31, 0x3a, 0x8, 0x31, 0x30, 0x2e, 0x30, 0x2e,
 		0x30, 0x2e, 0x32, 0x40, 0xd2, 0x9, 0x48, 0xae, 0x2c, 0x50, 0x6, 0x58, 0xe8,
 		0x7, 0x60, 0xe8, 0x7, 0x68, 0xe8, 0x7, 0x70, 0xe8, 0x7, 0x78, 0x32, 0x80,
 		0x1, 0x32, 0x88, 0x1, 0x32, 0x90, 0x1, 0x32, 0xca, 0x1, 0xb, 0x31, 0x39,
 		0x32, 0x2e, 0x31, 0x36, 0x38, 0x2e, 0x30, 0x2e, 0x31, 0x8a, 0x2, 0x9, 0x31,
 		0x32, 0x37, 0x2e, 0x30, 0x2e, 0x30, 0x2e, 0x31, 0x90, 0x2, 0x83, 0x25}
-	msg2InBytes = []byte{0x86, 0x1, 0x10, 0x1, 0x18, 0xd2, 0x9, 0x32, 0x19, 0x32,
+	msg2InBytes = []byte{0x10, 0x1, 0x18, 0xd2, 0x9, 0x32, 0x19, 0x32,
 		0x30, 0x30, 0x31, 0x3a, 0x30, 0x3a, 0x33, 0x32, 0x33, 0x38, 0x3a, 0x64,
 		0x66, 0x65, 0x31, 0x3a, 0x36, 0x33, 0x3a, 0x3a, 0x66, 0x65, 0x66, 0x62,
 		0x3a, 0x19, 0x32, 0x30, 0x30, 0x31, 0x3a, 0x30, 0x3a, 0x33, 0x32, 0x33,
```

Comment:
@srikartati when I changed to marshal function from `google.golang.org/protobuf/proto`, the marshal result and encoding result is a little bit different. Do you know the reason for that?

Reply:
@zyiou Sorry for the delayed response.
The difference is because the `encodedMessage` method prefixes the message with length whereas Marshal doesn't do that. https://godoc.org/github.com/golang/protobuf/proto#Buffer.EncodeMessage

Reply:
Got it. Do I need to prepend the length to current bytes to keep things same as before? Because `google.golang.org/protobuf/proto` does not have the exact same function as `EncodeMessage` in previous version.

Reply:
Looked at this more closely. This depends on the Kafka decoder functionality--some seem to expect length-delimited messages and some do not. I did not consider both options in my previous PR as we have a mock consumer in tests, there was no issue. 
However, it is better to support both with an input bool option "KafkaDelimitMsgWithLen" by defining a default value.

Reply:
Got it. Will add it in this patch since it is part of fixing for golangci-lint. Thanks!

Reply:
Are you thinking of fixing this in separate PR?

Reply:
Yes, when we cherry-pick the fixes from v0.4.3 to master branch, we can fix that. Since now the 0.4.0 branch does not have the Kafka commit.

Reply:
sounds good.